R/utils.R

```diff
@@ -20,18 +20,39 @@ base_pkgs <- function() {
 }
 
 #' @importFrom pkgcache ppm_snapshots
-get_ppm_snapshot_by_date <- function(date) {
+#' @examplesIf Sys.getenv("R_USER_CACHE_DIR", "") != ""
+#' get_ppm_snapshot_by_date(NA)
+#' get_ppm_snapshot_by_date("2023-08-01")
+#' get_ppm_snapshot_by_date(Sys.Date() + 10)
+get_ppm_snapshot_by_date <- function(date,
+                                     fallback_repo = file.path(pkgcache::ppm_repo_url(), "latest")) {
+  if (is.na(date) || is.infinite.POSIXlt(date)) {
+    return(fallback_repo)
+  }
+
   snaps <- pkgcache::ppm_snapshots()
-  res <- as.character(as.Date(head(
+  res <- as.character(as.Date(utils::head(
     snaps[as.Date(snaps$date) > as.Date(date), "date"],
     1
   )))
-  if (length(res) == 0) stop(sprintf("Cannot find PPM snapshot for date after %s.", as.character(date)))
-  res
+  if (length(res) == 0) {
+    warning(sprintf(
+      paste0(
+        "Cannot find PPM snapshot for date after %s.",
+        " Will use latest ppm snapshot instead."
+      ),
+      as.character(date)
+    ))
+    return(fallback_repo)
+  }
+  parse_ppm_url(res)
 }
 
 #' @importFrom pkgcache ppm_repo_url
-parse_ppm_url <- function(snapshot) {
+parse_ppm_url <- function(snapshot = NA) {
+  if (is.na(snapshot)) {
+    return(NULL)
+  }
```

Comment:
[low priority]
can we somehow get rid of `NA`? It's ambigious and could hold a lot of different information.
I spotted that you were trying to add `-Inf` to max, there is also a fallback functionality right above so it seems that you are going into the direction of fixing inputs to that function which is a nice move btw. This makes me thinking whether we still need this if here.